DEVINFO.md

```diff
@@ -0,0 +1,174 @@
+
+A good PR review sets both the submitter and reviewers up for success.  It minimizes the time required by both parties to get the code into an acceptable state, without sacrificing quality or safety.  Unlike most other software development practices, _safety_ is the primary concern.  A PR can and will be delayed or closed if there is any concern that it will lead to unintended consensus-breaking changes.
+
+This document describes some best practices on how to create and review PRs in this context.  The target audience is people who have commit access to this repository (reviewers), and people who open PRs (submitters).  This is a living document -- developers can and should document their own additional guidelines here.
+
+This document is formatted like a checklist.  Each paragraph is one goal or action item that the reviewer and/or submitter must complete.  The **key take-away** from each paragraph is bolded.
+
+## Reviewer Expectations
+
+The overall task of a reviewer is to create an **acceptance plan** for the submitter.  This is simply the list of things that the submitter _must_ do in order for the PR to be merged.  The acceptance plan should be coherent, cohesive, succinct, and complete enough that the reviewer will understand exactly what they need to do to make the PR worthy of merging, without further reviews.  The _lack of ambiguity_ is the most important trait of an acceptance plan.
+
+Reviewers should **strive to complete the review in one round**.  The reviewer should provide enough detail to the submitter that the submitter can make all of the requested changes without further supervision.  Whenever possible, the reviewer should provide all of these details publicly as comments, so that _other_ reviewers can vet them as well.  If a reviewer _cannot_ complete the review in one round due to its size and complexity, then the reviewer may request that the PR be simplified or broken into multiple PRs.
+
+Reviewers should aim to **perform a reviewer in one sitting** whenever possible.  This enables a reviewer to time-box their review, and ensures that by the time they finish studying the patch, they have a complete understanding of what the PR does in their head.  This, in turn, sets them up for success when writing up the acceptance plan.  It also enables reviewers to mark time for it on their calendars, which helps everyone else develop reasonable expectations as to when things will be done.
+
+Code reviews should be timely.  A PR review should begin no more than **2 business days** after the PR is submitted.  The `develop` and `next` branches in particular often change quickly, so letting a PR languish only creates more merge work for the submitter.  If a review cannot be begun within 2 business days, then the reviewers should **tell the submitter when they can begin**.  This gives the reviewer the opportunity to keep working on the PR (if needed) or even withdraw and resubmit it.
```

Comment:
> then the reviewers should **tell the submitter when they can

Can we be more specific about how we should use GitHub tooling to communicate this effectively?

Reply:
I'd prefer to leave that unspecified.  The only thing that matters is that the reviewer sets the submitter's expectations accordingly.